core-data/index.js

```diff
@@ -10,12 +15,31 @@ import reducer from './reducer';
 import * as selectors from './selectors';
 import * as actions from './actions';
 import * as resolvers from './resolvers';
+import entities from './entities';
+
+const entityResolvers = entities.reduce( ( memo, { kind, name } ) => {
+	const kindPrefix = kind === 'root' ? '' : upperFirst( camelCase( kind ) );
```

Comment:
I would love to see the following method to avoid code duplication and ensure that both selector and resolver names always match:
```js
const getMethodName = ( kind, name ) => {
    const kindPrefix = kind === 'root' ? '' : upperFirst( camelCase( kind ) );
    const nameSuffix = upperFirst( camelCase( name ) );
    return `get${ kindPrefix }${ nameSuffix }`;
};
```
This logic is quite complicated and essential to have it working properly for all selectors so it would be great to have a simple unit tests which ensures it never regresses.

Reply:
Good suggestion, updating.

Reply:
or:

```js
const generateEntityMethods = ( source) => {
	const entityResolvers = entities.reduce( ( memo, { kind, name } ) => {
		const kindPrefix = kind === 'root' ? '' : upperFirst( camelCase( kind ) );
		const nameSuffix = upperFirst( camelCase( name ) );
		return {
			...memo,
			[ `get${ kindPrefix }${ nameSuffix }` ]: ( state, key ) => source.getEntityRecord( state, kind, name, key ),
		};
	}, {} );

const entityResolvesr = generateEntityMethods( resolvers );
const entitySelectors = generateEntityMethods( selectors );
```